### PR TITLE
MM-33544 is_following prop in getPosts API methods

### PR DESCRIPTION
**Pull request by reflog**
#### Summary

added is_following to getPosts in collapsed thread mode
(i know the amount of changes in this PR is insane, but I needed to pass userID to store functions, since the result of the fetch depends on who's asking about the post)

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-33544

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: app/post_test.go [hunk 1]
```diff
@@ -1943,7 +1943,7 @@ func TestCollapsedThreadFetch(t *testing.T) {
 			Message:   "root post",
 		}, channel, false, true)
 		require.Nil(t, err)
-
+		time.Sleep(1)
```
[the remark this answers is not in the document]
Comment (reflog) [reply app/post_test.go:1946]: oops. leftover from testing :) removed

File: model/post.go [hunk 2]
```diff
@@ -101,6 +101,7 @@ type Post struct {
 	ReplyCount   int64         `json:"reply_count" db:"-"`
 	LastReplyAt  int64         `json:"last_reply_at" db:"-"`
 	Participants []*User       `json:"participants" db:"-"`
+	IsFollowing  bool          `json:"is_following" db:"-"` // for root posts in collapsed thread mode indicates if the currect user is following this thread
```
Comment (agnivade) [inline model/post.go:104]: current or correct?

**Review by agnivade — APPROVED** [review]

File: model/post.go (hunk 2 above)
Comment (reflog) [reply model/post.go:104]: @agnivade I said "currect" and I stand by it! :)
I'll fix the typo.

**Review by streamer45 — APPROVED** [review]
:+1:

**Comment by reflog** [comment]
@ogi-m general smoke testing around posts/replies, just to check that I didn't break something unexpected

**Review by ogi-m — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**